http-server-netty/src/main/java/io/micronaut/http/server/netty/handler/Compressor.java

```diff
@@ -185,4 +193,80 @@ private enum Algorithm {
             this.contentEncoding = contentEncoding;
         }
     }
+
+    static final class Session {
+        private final EmbeddedChannel compressionChannel;
+        private boolean finished = false;
+
+        private Session(ChannelHandlerContext ctx, ChannelHandler handler) {
+            compressionChannel = new EmbeddedChannel(
+                ctx.channel().id(),
+                ctx.channel().metadata().hasDisconnect(),
+                ctx.channel().config(),
+                handler
+            );
+        }
+
+        void push(ByteBuf data) {
+            if (finished) {
+                throw new IllegalStateException("Compression already finished");
+            }
+            if (data.isReadable()) {
+                compressionChannel.writeOutbound(data);
+            } else {
+                data.release();
+            }
+        }
+
+        void finish() {
+            if (!finished) {
+                compressionChannel.finish();
+                finished = true;
+            }
+        }
+
+        void discard() {
+            if (!finished) {
+                try {
+                    compressionChannel.finishAndReleaseAll();
+                } catch (DecompressionException ignored) {
```

Comment:
should not we log this exception?

Reply:
this happens when a user closes a connection so we want to avoid log spam from half-finished data